version: '3.9'

services:
  pheme-db:
    image: postgres:14.5
    restart: always
    volumes:
      - pheme:/var/lib/postgresql/data
    ports:
      - 5432:5432
    networks:
      - pheme-network
    environment:
      - DATABASE_HOST=${DATABASE_HOST}
      - DATABASE_PORT=${DATABASE_PORT}
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DB=${POSTGRES_DB}

  golang-pipeline:
    depends_on:
      pheme-db:
        condition: service_started
    volumes:
      - .:/app
    ports:
      - 8000:8000
    networks:
      - pheme-network
    environment:
      - DATABASE_HOST=${DATABASE_HOST}
      - DATABASE_PORT=${DATABASE_PORT}
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DB=${POSTGRES_DB}
      - SERVER_HOST=${SERVER_HOST}
      - SERVER_PORT=${SERVER_PORT}
    build:
      context: ..
      dockerfile: ./ci/pheme_auth.Dockerfile
    tty: true
    restart: on-failure

volumes:
  pheme:

networks:
  pheme-network:
    driver: bridge